bip-XXXX.mediawiki

```diff
@@ -0,0 +1,93 @@
+<pre>
+  BIP: XXXX
+  Layer: Applications
+  Title: DNS Payment Instructions
+  Author: Matt Corallo <bipxxx@bluematt.me>
+          Bastien Teinturier <bastien@acinq.fr>
+  Comments-Summary: No comments yet.
+  Status: Draft
+  Type: Standards Track
+  Created: 2024-02-10
+</pre>
+
+==Abstract==
+This BIP proposes a standard format for encoding [[bip-0021.mediawiki|BIP 21]] URI schemes in DNS TXT records.
+
+==Motivation==
+Various Bitcoin and other cryptocurrency applications have developed human-readable names for payment instructions over time, with marketplace adoption signaling strong demand for it from users.
+
+The DNS provides a standard, global, hierarchical namespace mapping human-readable labels to records of various forms. Using DNSSEC, the DNS provides cryptographic guarantees using a straightforward PKI which follows the hierarchical nature of the DNS, allowing for stateless and even offline validation of DNS records from a single trusted root.
+
+Further, because DNS queries are generally proxied through ISP-provided or other resolvers, DNS queries usually do not directly expose the queryer's IP address. Further, because of the prevalence of open resolvers, the simplicity of the protocol, and broad availability of DNS recursive resolver implementations, finding a proxy for DNS records is trivial.
+
+Thus, using TXT records to store Bitcoin payment instructions allows for human-readable Bitcoin payment destinations which can be trivially verified on hardware wallets and which can be resolved relatively privately.
+
+==Specification==
+
+=== General rules for handling ===
+Bitcoin wallets MUST NOT prefer to use DNS-based resolving when methods with explicit public keys are available. In other words, if a standard Bitcoin address or direct BIP 21 URI is available or would suffice, Bitcoin wallets MUST prefer to use that instead.
+
+=== Records ===
+Payment instructions are indexed by both a user and a domain. Instructions for a given `user` and `domain` are stored at `user`.user._bitcoin-payment.`domain` in a single TXT record.
```

Comment:
I don't quite understand the scheme here. What is the purpose of the "user" label? (Future protocol expansions where something besides "users" can be identified?). Also why not order the labels with the underscores more like existing schemes for example, TLSA:

`_443._tcp.mailhardener.com TLSA ...`

So I'd expect something like:

`_bitcoin-payment.matt.mattcorallo.com TXT ...`

Reply:
(see also https://github.com/james-stevens/wallet-ids-in-dns)

Reply:
@pinheadmz 

> I'd expect something like:
>
> `_bitcoin-payment.matt.mattcorallo.com TXT ...`

I'm not DNS expert, but the Wikipedia article about [wildcard DNS entries](https://en.wikipedia.org/wiki/Wildcard_DNS_record) says, "A wildcard DNS record is specified by using a `*` as the leftmost label (part) of a domain name, e.g. `*.example.com`."  I think @TheBlueMatt wants to allow wildcards for the username portion for LN (see the proposed BIP21 parameter `omlookup` (onion message lookup) in [this commit](https://github.com/lightning/bolts/pull/1136/commits/5809dfd1fce7fe1233812acb25bd37b908318968)).  If only the leftmost label can be a wildcard, then `_bitcoin-payment.*.mattcorallo.com` wouldn't be allowed.